src/group_theory/free_abelian_group.lean

```diff
@@ -380,6 +384,47 @@ instance [monoid α] : ring (free_abelian_group α) :=
   .. free_abelian_group.add_comm_group α,
   .. free_abelian_group.semigroup α }
 
+variable {α}
+
+def of_mul_hom [monoid α] : α →* free_abelian_group α :=
+{ to_fun := of,
+  map_one' := rfl,
+  map_mul' := of_mul }
+
+@[simp] lemma of_mul_hom_coe [monoid α] : (of_mul_hom : α → free_abelian_group α) = of := rfl
+
+/-- If `f` preserves multiplication, then so does `lift f`. -/
+def lift_monoid {R : Type*} [ring R] [monoid α] : (α →* R) ≃ (free_abelian_group α →+* R) :=
```

Comment:
Should there be simp-lemmas for this def?

Reply:
Perhaps, but this exists primarily to enable the other two lifts in this PR - previously the proofs here were duplicated in both places.

What simp lemmas would you want?

Reply:
I've added the three most interesting ones I could think of.